operator/README.md

```diff
@@ -0,0 +1,173 @@
+<!-- TOC -->
+- [PMEM-CSI Operator](#pmem-csi-for-kubernetes)
+  - [About](#about)
+  - [Usage](#usage)
+    - [Pre-requisites](#pre-requisites)
+    - [Deploy PMEM-CSI operator](#deploy-pmem-csi-operator)
+    - [Deployment CRD API](#deployment-crd-api)
+  - [Example Deployment](#example-deployment)
+
+<!-- /TOC -->
+# PMEM-CSI-Operator
+
+## About
+
+Kubernetes operator for [PMEM-CSI](https://github.com/intel/pmem-csi) driver deployment. Using this operator one can easily deploy and manage PMEM-CSI driver in a Kubernetes cluster.
+
+The PMEM-CSI operator is based on the CoreOS [operator-sdk](https://github.com/operator-framework/operator-sdk) tools and APIs.
+
+## Usage
+
+### Pre-requisites
+
+A running Kubernetes cluster with one or more nodes installed with PMEM hardware(NVDIMM). The persistent memory on these nodes must be [pre-provisioned](../README.md#persistent-memory-pre-provisioning).
+
+### Deploy PMEM-CSI operator
+
+<!-- The assumptions:
+  a) pmem-csi-operator binary is part of released intel/pmem-csi-driver image and no additional steps required to build the images
+-->
+
+```sh
+$ kubectl create -f https://raw.githubusercontent.com/intel/pmem-csi/operator/operator/deploy/operator.yaml
+```
+
+### Deployment CRD API
+
+Current PMEM-CSI Deployment object supports below API:
+
+#### Deployment
+
+|Field | type | Description |
+|---|---|---|
+| apiVersion | string  | `pmem-csi.intel.com/v1alpha1`|
+| kind | string | `Deployment`|
+| metadata | [ObjectMeta](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status) | Standard objects metadata |
+| spec | [DeploymentSpec](#deployment-spec) | Specification of the desired behavior of the deployment |
+
+#### DeploymentSpec
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| driverName | string | Unique CSI driver name to use | `pmem-csi.intel.com` |
+| image | string | PMEM-CSI docker image name used for the deployment | `intel/pmem-csi-driver:canary` |
+| provisionerImage | string | [CSI provisioner](https://github.com/kubernetes-csi/external-provisioner) docker image name | `quay.io/k8scsi/csi-provisioner:v1.2.1` |
```

Comment:
Same here: the default may change between releases. Instead of copying that information, just point the user to where they can find the latest information.

Speaking of that, how does the operator know which version to use? It may depend on the Kubernetes version. Does it keep a builtin copy of the "deploy" files and then modify those? That's currently our canonical source of truth for how to deploy the driver for the different supported Kubernetes versions.

Reply:
> Speaking of that, how does the operator know which version to use? It may depend on the Kubernetes version

Though not implemented to support < 1.16 Kubernetes versions part of this PR, It supposed to [determine the running cluster version](https://github.com/intel/pmem-csi/pull/525/commits/7187383845b38f7495d47588b328a0bcb76c5fde) and prepares the appropriate driver deployment manifests.